### ISSUE TITLE
Implementation of IStructuralEquatable.GetHashCode in Array class consider just element with 0 index

### ISSUE THREAD

**Issue by etomm** [dotnet/corefx#4379]
This is the implementation:

```
int IStructuralEquatable.GetHashCode(IEqualityComparer comparer)
{
    if (comparer == null)
    {
        throw new ArgumentNullException("comparer");
    }
    int num = 0;
    for (int i = (this.Length >= 8) ? (this.Length - 8) : 0; i < this.Length; i++)
    {
        num = CombineHashCodes(num, comparer.GetHashCode(this.GetValue(0)));
    }
    return num;
}
```

As you can see the CombineHashCodes keeps combining the hashcode of element with 0 index. 

This makes array as new[]{"aa","bb"} and new[]{"aa","cc"} to have the same hashcode. And the probability that there are array with the first element equal is usually higher.

**Comment by etomm** [comment]
Ok, I could find the file right now and I see that it has been corrected, I hope it enter a fix release for version 4.5

**Comment by ianhays** [comment]
This has been resolved. The code is now:
```
int IStructuralEquatable.GetHashCode(IEqualityComparer comparer) {
    if (comparer == null)
        throw new ArgumentNullException("comparer");
    Contract.EndContractBlock();

    int ret = 0;

    for (int i = (this.Length >= 8 ? this.Length - 8 : 0); i < this.Length; i++) {
        ret = CombineHashCodes(ret, comparer.GetHashCode(GetValue(i)));
    }

    return ret;
}
```
